sympy/physics/mechanics/tests/test_joint.py

```diff
@@ -0,0 +1,347 @@
+                                                  omega*x_x*parent.frame.y)
+    assert parent.frame.dcm(child.frame) == Matrix([
+        [cos(theta), -sin(theta), 0],
+        [sin(theta),  cos(theta), 0],
+        [0, 0, 1]])
+
+
+def test_planar_joint_arguments():
+    parent = Body('parent')
+    child = Body('child')
+    l1 = Symbol('l1')
+    l2 = Symbol('l2')
+    planar_joint = PlanarJoint('planar_joint', parent, child, child_axis='y',
+                               parent_distance=l1, child_distance=l2)
+    assert planar_joint.name == 'planar_joint'
+    assert planar_joint.parent == parent
+    assert planar_joint.child == child
+    assert planar_joint.parent_axis == parent.frame.z
+    assert planar_joint.child_axis == child.frame.y
+    assert planar_joint._parent_joint_location == l1 * parent.frame.z
+    assert planar_joint._child_joint_location == l2 * planar_joint.child_axis
+
+
+def test_spherical_joint():
+    parent = Body('parent')
+    child = Body('child')
+    l = Symbol('l')
+    spherical_joint = SphericalJoint('spherical_joint', parent, child,
+                                     child_point_pos=(0, l, 0))
+
+    assert spherical_joint.name == 'spherical_joint'
+    assert spherical_joint.parent == parent
+    assert spherical_joint.child == child
+    assert spherical_joint._parent_joint_location == Vector(0)
+    assert spherical_joint._child_joint_location == l * child.frame.y
+
+    alpha = dynamicsymbols('spherical_joint_alpha')
+    beta = dynamicsymbols('spherical_joint_beta')
+    gamma = dynamicsymbols('spherical_joint_gamma')
+    alphad = dynamicsymbols('spherical_joint_alpha', 1)
+    betad = dynamicsymbols('spherical_joint_beta', 1)
+    gammad = dynamicsymbols('spherical_joint_gamma', 1)
+    omega_alpha = dynamicsymbols('spherical_joint_omega_alpha')
+    omega_beta = dynamicsymbols('spherical_joint_omega_beta')
+    omega_gamma = dynamicsymbols('spherical_joint_omega_gamma')
+
+    assert spherical_joint.coordinates == [alpha, beta, gamma]
+    assert spherical_joint.speeds == [omega_alpha, omega_beta, omega_gamma]
+    assert spherical_joint.kds == [alphad - omega_alpha, betad - omega_beta,
+                                   gammad - omega_gamma]
+
+    child_joint_point = spherical_joint.child_joint_point
+    parent_joint_point = spherical_joint.parent_joint_point
+    assert parent_joint_point.name == 'spherical_joint_parent_joint'
+    assert child_joint_point.name == 'spherical_joint_child_joint'
+    assert child_joint_point.pos_from(parent_joint_point) == Vector(0)
+    assert child_joint_point.pos_from(child.masscenter) == l * child.frame.y
+    assert parent_joint_point.pos_from(parent.masscenter) == Vector(0)
+    assert child.frame.ang_vel_in(parent.frame) == (omega_alpha * parent.frame.x +
+                                                    omega_beta * parent.frame.y +
+                                                    omega_gamma * parent.frame.z)
```

Comment:
Body fixed rotations are supposed to be: `omega_alpha * parent_frame.x + omega_beta * intermediate_frame.y + omega_gamma * child_frame.z`. It may be best to set the kinematical differential equations like so:

```python
omega_alpha = child.ang_vel_in(parent).dot(child.x)
omega_beta = child.ang_vel_in(parent).dot(child.y)
omega_gamma = child.ang_vel_in(parent).dot(child.z)

child.set_vel(parent_frame, omega_alpha * child_frame.x + omega_beta * child_frame.y + omega_beta * child_frame.z)
```

What you currently have is likely incorrect unless you defined the generalized speeds to be in terms of the parent frame.